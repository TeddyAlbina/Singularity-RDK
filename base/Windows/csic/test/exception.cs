using System;
class test {
	static void F() {
		try {
			G();
		} catch (Exception e) {
			Console.WriteLine("Exception in F: " + e.Message);
			e = new Exception("F");
			//throw;		// re-throw
			throw e;	// throw a new exception
		} catch {
			;
		} finally {
			Console.WriteLine("F finally block");
		}
	}
	static void G() {
		throw new Exception("G");
	}
	public static void Main() {
		try {
			F();
		} catch (Exception e) {
			Console.WriteLine("Exception in Main: " + e.Message);
		}
	}
}