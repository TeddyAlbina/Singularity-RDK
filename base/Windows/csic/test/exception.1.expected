Exception in F: G
F finally block
Exception in Main: F
